core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -92,6 +94,20 @@ class KryoSerializer(conf: SparkConf)
       new KryoOutput(bufferSize, math.max(bufferSize, maxBufferSize))
     }
 
+  @transient
+  private lazy val factory: KryoFactory = new KryoFactory() {
+    override def create: Kryo = {
+      newKryo()
+    }
+  }
+
+  @transient
```

Comment:
Back on this item now, if this is no longer a lazy val, and it's transient, how does it get set again after this object itself is serialized? I'm mostly wondering if this is required now.

Reply:
Yes, you are correct, I don't think this will work